packages/dds/tree/src/shared-tree/schematizingTreeView.ts

```diff
@@ -219,6 +229,44 @@ export class SchematizingSimpleTreeView<
 		return this.view;
 	}
 
+	/**
+	 * Run a transaction which applies one or more edits to the tree as a single atomic unit.
+	 */
+	public runTransaction<TResult>(
+		params: TransactionParams<TResult>,
+	): TransactionStatus<TResult> {
```

Comment:
I think we may want to decouple the type that is returned by `TransactionParams.transaction()` from the type returned by `runTransaction`.

For example, I don't see much value in including the constraints in the object that is returned by `runTransaction`.
In the future, we'll likely also want to add a bunch of useful things to the object that is returned by `runTransaction` (like something to keep track of whether the associated commit has been sequenced) and that's definitely not something that the user could supply in the object they return from `TransactionParams.transaction()`.